### ISSUE TITLE
Bundled software versions as a potential indicator of new firmware compatibility?

### ISSUE THREAD

**Issue by Danimations** [Effix/LedBadge#16]
Hey @Effix . 

I have some news. I've received quite a few badges from different suppliers now (via eBay and AliExpress) and some of them have clear external differences. I've also received quite a few different loader programs for Windows too, and have discovered the merits and limits of most of them.

Aside from the ability to display half-tones that your firmware UNIQUELY unlocks, I've been able to get good results out of Mini LED Display v9.17, v9.92 and to a lesser extent, v10.2. I've overcome the apparent (but falsely so) 8-frame limitation by learning how the software works. While a little counter intuitive, I can now program plenty of screen time. I'm yet to find the storage limit of any of the badges I've received.

There was one anomaly among the various software received. "HD Smart" is quite a different program, that works with art in 11x11 pixel blocks. These include loading animated .gif files (rather than manually loading a series of .bmp files or similar), which is a big plus.

I wanted to ask you what software was bundled with your b1236 and b1248 badges? I suspect that could be a rough indicator of what components are inside the chassis. Version numbers of the compatible supplied software would be good to know. 

I'd prefer not to open up _too_ many more of the badges I've received, as I'd like to resell some of them as artworks to recover some of the costs sunk into this project already.

**Comment by Effix** [comment]
I don't have the software anymore, but I want to say it was v5 or v6.

**Comment by Danimations** [comment]
Do you recall whether the software title was "Mini LED Display Vx" or similar? I have a program here called "LED Mini Board - Badge English V5" which might be a match? It's very limiting, and wouldn't allow me to animate full screen 1-bit images the way I can with Mini LED Display V9.17 or upwards. I think I found that title by scrounging online... I don't believe I have any hardware compatible with "LED Mini Board".

**Comment by Effix** [comment]
I believe it was the Mini Board one - it was very limited. Like 6 messages and only static images.

**Comment by Danimations** [comment]
I found a China-based manufacturer who can ship 12x36s with the LED Mini Board software. I turned down their offer, as the software appeared to be too limiting, and I couldn't rely on them being compatible with your work. 

I bought two dozen 11x33's instead, which _are_ compatible with Mini LED Display V9.92, which will allow me to animate in monochrome to my heart's content, without having to crack open the backs of badges. I just pay for the privilege in one lost row and three lost columns of resolution.

Perhaps when I lodge my next order, I could get one of their 12x36s to open up, and see if it compares with yours.

**Comment by Danimations** [comment]
Does this look like the program that came with your badges, @Effix ?

![led mini board - badge english v5](https://user-images.githubusercontent.com/39525759/42134598-4192d7aa-7d7e-11e8-9a31-945899e557f8.PNG)

**Comment by Effix** [comment]
Yeah, that looks familiar.

I wouldn't necessarily assume that a badge shipped with that software is guaranteed to be compatible. Any hardware revision that implements the same protocol would work.

**Comment by Danimations** [comment]
Yeah, I figured it _could_ be a clue that might suggest compatibility, but wouldn't be a given. 

The above program was the obvious candidate following your description of the software that was bundled with your badges. 

The more useful software (Mini LED 9.17+) has 8 universal slots rather than this 6/2 configuration, each of which can be used to hold a string of text, one long (wide) image, or a sequence/set of images (Mini LED 9.92+ only). By choosing "Animation" as the effect for any of these slots, a new frame is assumed every 48 or 36 pixels when the badge plays back any image data. It flicks through the "frames" in rapid succession, creating the illusion of movement. The range of playback frame rates is quite versatile and useful.

**Comment by madcock** [comment]
> I have some news. I've received quite a few badges from different suppliers now (via eBay and AliExpress) and some of them have clear external differences. I've also received quite a few different loader programs for Windows too, and have discovered the merits and limits of most of them.

Hi, sorry for the necropost, but I appear to be in a somewhat similar situation you were in, just 6 years in the future. ;)

I got a few badges, did some research online, and decided to do a little art project for some of my local pinball friends. Then I proceeded to buy several badges from different sellers to get various colors at better prices, and ended up with a set of different hardware. Most of them have been 11x44, and work with what appears to be updated software to what you received (or fond) within this thread. Most of my badges are still usable with their software, though getting my artwork onto all of them is going to be more of a challenge than I anticipated. In case you're curious, here's [my work in progress website](https://madcock.github.io/PixelPins/) that has various custom animations I've created, or adapted from some games.

My question is this... Do you still have copies of the old badge software, and would you be willing to share? :)

Currently I've got these, which all have various pros and cons, but are at least capable of what I assume was the same trick you discovered years ago -- using a BMP that is basically 11 tall x 44*N wide where N is the number of frames of the animation, and then setting the animation flag so it shows them in rapid succession.
![image](https://github.com/Effix/LedBadge/assets/2106838/b00832d0-5bc7-4056-87fa-c68b527e57ff)
![image](https://github.com/Effix/LedBadge/assets/2106838/20174215-4b13-46ca-8bbc-e3b8f8b969cd)
![image](https://github.com/Effix/LedBadge/assets/2106838/9c459848-ec4d-48bd-8357-bced141e7fe2)

But I just received a few more badges that only linked to an Android app (a very poor quality app, as is true with most of the apps for these things -- the Windows software isn't great either, but it's generally way better than the apps!) None of the software I've tried, or the python projects I've found online can connect to this different badge type. And I'm hoping maybe it's old, and perhaps compatible with one of the types you ended up with back in the day... There's one thing that distinguishes this one from the others I've seen -- it doesn't have the "coffee" animation when charging, but instead has a flashing lightening bolt  with the battery image on screen.

I know it's a long shot, but I figured I'd ask. Currently, the only thing I can put on this new set of badges is text and what I guess I'd call "emoji" images -- they default to 11x11, can be mixed with the text, but can be made to grow to a max of 11x128 in the app. But 128 wide is not nearly enough for anything but simple animations. And the image importer in the app is so poor it corrupts/dithers/resizes everything loaded, so it would require remaking every single image pixel by pixel, by hand, and with only 2 frames total of animation. :( The badge does get recognized as "something" when plugged into Windows, and I expect there is some software, somewhere, that might work with it. I had no luck with Linux options, though they work fine for my other badges. Any pointers or suggestions from your struggles in the past would be appreciated! I followed your journey here in comments. Hope your project(s) worked out in the end!

**Comment by J0w03L** [comment]
> Does this look like the program that came with your badges, @Effix ?
> 
> ![led mini board - badge english v5](https://user-images.githubusercontent.com/39525759/42134598-4192d7aa-7d7e-11e8-9a31-945899e557f8.PNG)

I can confirm this looks more or less identical to the program that came with my 48x12 badge. I uploaded the software that came with it to Archive.org a while ago, so you can compare my version and packaging with yours should you wish: https://archive.org/details/a16nf-drivers-6.40

As for getting Linux support on these things, I'm not sure about the official software/firmware, but I have managed to write my own software for these on a version of the custom firmware from this repository that does work on Linux. Here's me playing Bad Apple with it: https://youtu.be/pU4XbttlwYc

**Comment by J0w03L** [comment]
Here's a fun fact about the .bge file format that that program uses. If you hex-edit the speed to something invalid, you can achieve a much higher FPS than what the program will ever allow you to do. I used to upload a "spritesheet" through the logo feature and use that hack to get smooth animations on there. Maybe you could try doing something similar on yours, @madcock? Hope this helps!

**Comment by Effix** [comment]
Unfortunately, I am unable to find my original disk. It was one of those mini cds like what J0w03L archived. I'm not even sure that version would left you flip through frames of a sprite sheet or if it was just a sliding image. Probably the best you could do with that version is flip between the two image slots it had.
